gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/internal/PluginEventListener.java

```diff
@@ -197,18 +197,22 @@ public int compare(Plugin o1, Plugin o2) {
         }
```

Comment:
I would remove the SecretProviderPluginComparator.
To do so, create a constructor to PluginComparator to get `secrets.loadFirst` property.
Then move the comparison logic and set it where `return 0` is today.
If plugin are of the same type, we need to put up the stack the `loadFirst` one. That would easier to understand imo.